src/general/world_effects/MarineSnowDensityEffect.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Adjusts marine snow densities in patches to correlate with populations of species (so that it makes sense for
+///   it to appear only after there are microbes that are assumed to have died to create the marine snow)
+/// </summary>
+public class MarineSnowDensityEffect : IWorldEffect
+{
+    private const string MarineSnowConfiguration = "marineSnow";
+    private const string TemplateBiomeForMarineSnow = "mesopelagic";
+    private const float DepthDifferenceForMarineSnowFalling = 10;
+    private const float MarineSnowPatchBottomExtraForFalling = 200;
+
+    private readonly HashSet<OrganelleDefinition> availableOrganelles = new();
+
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    public MarineSnowDensityEffect(GameWorld targetWorld)
+    {
+        this.targetWorld = targetWorld;
+    }
+
+    public void OnRegisterToWorld()
+    {
+    }
+
+    public void OnTimePassed(double elapsed, double totalTimePassed)
+    {
+        var templateBiome = SimulationParameters.Instance.GetBiome(TemplateBiomeForMarineSnow);
+        var fullMarineSnowConfig = templateBiome.Conditions.Chunks[MarineSnowConfiguration];
+
+        foreach (var key in targetWorld.Map.Patches.Keys)
```

Comment:
Isn't better to use ".Values" insted of ".Keys" here?

Reply:
That's probably the case. I copy-pasted this from another IWorldEffect and just left the skeleton before starting to add the stuff specific for this usage. I'll swap this out to `.Values` tomorrow before merging.

Reply:
Actually I still had the branch checked out so I committed a change now.